src/api/hooks/useNavigateWithParams.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  createSearchParams,
+  generatePath,
+  useNavigate,
+  useSearchParams,
+} from "react-router-dom";
+import {defaultNetworkName} from "../../constants";
+
+export function useNavigateWithParams() {
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+
+  const networkName = searchParams.get("network") || defaultNetworkName;
```

Comment:
what happens if url has `?network=zihan`?

Reply:
it'll bring us to `?network=mainnet`